llvm/lib/IR/RepoDefinition.cpp

```diff
@@ -90,24 +90,37 @@ const Constant *getAliasee(const GlobalAlias *GA) {
 
 template <typename GlobalType>
 void ModuleHashGenerator::calculateGOInfo(const GlobalType *G) {
-  GOInfo Result = calculateDigestAndDependenciesAndContributedToGVs(G);
-  // ContributedToGVs of an object is used to update other objects'
-  // contributions. For example, if the ContributedToGVs of function `foo`  are
-  // global variables `g` and `q`, the function `foo` is in the contributions of
-  // `g` and `q`.
-  // ContributedToGVs[`foo`] = [`g`, `q`]
-  // ====> Contributions[`g`] = [`foo`],
-  //       Contributions[`q`] = [`foo`]
-  for (auto &GO : Result.Contributions) {
+  GOInfo GInitInfo = calculateDigestAndDependenciesAndContributions(G);
+#ifndef NDEBUG
+  LLVM_DEBUG(dbgs() << "\nGO Name:" << G->getName() << "\n");
+  LLVM_DEBUG(GInitInfo.dump(true));
+#endif
+  GOInfo &GInfo = GOIMap[G];
+  GInfo.InitialDigest = std::move(GInitInfo.InitialDigest);
+  // Move G's initial dependencies(`GInitInfo`) into its information map
+  // (`GInfo`).
+  std::move(GInitInfo.Dependencies.begin(), GInitInfo.Dependencies.end(),
+            std::back_inserter(GInfo.Dependencies));
```

Comment:
If there a reason why we _move_ the dependencies here, but _copy_ the contributions the second half of this function? The members of GOVec are `GlobalObject *`, so the move would degrade to a copy anyway.

Reply:
Good point, the second part has been changed from `copy` the contributions to `move` them. Fixed in commit cca1f2d.

Reply:
> the second part has been changed from copy the contributions to move them. Fixed in commit cca1f2d.

I’d go the other way. As I mentioned, move() does essentially nothing to a pointer: fundamental types don’t have move constructors. Using it here is rather incongruous and what sparked my original question.

Reply:
Thanks Paul, fixed in 31096559362.